crates/driver/src/infra/api/routes/solve/dto/auction.rs

```diff
@@ -6,23 +6,25 @@ use {
     itertools::Itertools,
     serde::Deserialize,
     serde_with::serde_as,
-    std::{collections::HashMap, str::FromStr},
+    std::collections::HashMap,
 };
 
 impl Auction {
     pub fn into_domain(self) -> Result<competition::Auction, Error> {
         Ok(competition::Auction {
-            id: Some(FromStr::from_str(&self.id).map_err(|_| Error::InvalidAuctionId)?),
+            id: Some((self.id as u64).into()),
             tokens: self
-                .tokens
+                .prices
                 .into_iter()
-                .map(|(address, token)| competition::auction::Token {
-                    decimals: token.decimals,
-                    symbol: token.symbol,
-                    address: address.into(),
-                    price: token.reference_price.map(Into::into),
-                    available_balance: token.available_balance,
-                    trusted: token.trusted,
+                // TODO: Populate currently hardcoded fields.
+                .map(|(key, value)| competition::auction::Token {
+                    decimals: None,
+                    symbol: None,
```

Comment:
I don't think this is necessarily part of the auction. To me, token metadata is "side-car" data (much like liquidity) which the driver "augments" before sending to the solver engine.